extensions/avro-rpc/runtime/src/main/java/org/apache/camel/quarkus/component/avro/rpc/spi/HttpServletResponseWithBytes.java

```diff
@@ -0,0 +1,252 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.component.avro.rpc.spi;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.Collection;
+import java.util.Locale;
+
+import javax.servlet.ServletOutputStream;
+import javax.servlet.WriteListener;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Empty implementation of {@link javax.servlet.http.HttpServletResponse}.
+ * Serves as a wrapper to be used for {@link org.apache.avro.ipc.ResponderServlet}.
+ */
+public class HttpServletResponseWithBytes implements HttpServletResponse {
+
+    private ServletOutputStreamWithBytes baos = new ServletOutputStreamWithBytes();
+
+    public byte[] getBytes() {
+        return baos.getBytes();
+    }
+
+    @Override
+    public ServletOutputStream getOutputStream() throws IOException {
+        return baos;
+    }
+
+    @Override
+    public void addCookie(Cookie cookie) {
+
+    }
+
+    // --------------------- following methods are not implemented and not used ----------------------------
+
+    @Override
+    public boolean containsHeader(String s) {
+        return false;
```

Comment:
Maybe better throw an UnsupportedOperationException here and below?

Reply:
Yes this would be better.

Reply:
done